### DISCUSSION TITLE
Oscilloscope modules

### DISCUSSION THREAD

**Discussion by gusmanb** [gusmanb/logicanalyzer#51] [Ideas]
I'm starting to get components to dessign the oscilloscope modules. The idea here is to have one or two ADC boards that can be triggered by a voltage level or a digital trigger.

For the digital trigger as we have 24 digital IO's available and the ADC's need 8 input channels for data and one clock signal I think we could have a digital trigger of up to 7 channels, or of course use a second analyzer and use the external trigger.

I have ordered a couple of ad9283 ADCs that can sample 8 bit at 100Msps, so it will give a good resolution with not an excessive price.

For the analog side of the device any help is welcome, I'm more a software/digital electronics guy than analog so any consideration, advice or help will be appreciated.

**Comment by phantomkingx** [comment]
Have you considered using the onboard pico ADCs which would provide 3 analog channels?  I understand they are limited to 500kS/s but that may be quite useful for many applications.  Although still not trivial the advantage would be just a firmware/software change to implement and it wouldn't use up any other channels.

Great work btw, just came across this project and see myself using the analyzer in the future.

**Comment by gusmanb** [comment]
Hi Simon!

Yes, that's something that I already have considered. It would be only useful for low frequency applications, with 500Ksps the maximum frequency that you would be sampling is 125Khz, and not in a very good fashion, only 4 samples per cycle will give you just an idea of the waveform, it would start to be useful at 60Khz~ and still I would not consider it "good".

In any case, I studied the possibility, at least for audio and low-speed I2C would be good enough but the Pico has a flaw in the ADCs which makes them useless for precise aplications, there are four concrete values that make the result to suddenly peak, you can check it on the [RP2040 datasheet](https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf), check errata RP2040-E11, so I discarded totally the idea.

Anyway, I already have received two [AD9283-100](https://www.analog.com/media/en/technical-documentation/data-sheets/ad9283.pdf) IC's and a [iCE40-ADC module](https://www.olimex.com/Products/FPGA/iCE40/iCE40-ADC/open-source-hardware), the AD9283-100 should be the ideal solution but will need a bit more tinkering by the user as it will need the assembling of the board, but the iCE40-ADC module is a pre-made solution that should work only by connecting 10 wires between the analyzer and the board. The best is that the firmware for both cases should be the same, so I think that having these two options available will be the best solution and will give an effective frequency of 25Mhz, which is something really useful.

Cheers!

**Reply by danielh186** [reply]
Do you know whether the flaw you are talking about got fixed in the RP2350?
Having 4 analog channels would be pretty cool. Even though I would not think of it as a replacement for a bench oscilloscope.

**Comment by phantomkingx** [comment]
Ah I see, always a fly in the ointment.

**Reply by p3t3rm0v** [reply]
It would be  great idea. I used adc pico to monitoring some fast traffic and it is poor quality as scoppy project... so some extra help as external quality adcs will be wellcome.
